source/Subsystem/EntityComponentSystem.h

```diff
@@ -18,10 +21,32 @@ class EntityComponentSystem
 	Entity* GetEntity(std::string aName);
 	void DestroyEntity(Entity* aEntity);
 
+	template<class T>
+	Component* CreateComponent(Entity* aEntity)
+	{
+		Component* componentToAdd = static_cast<Component*>(new T(aEntity));
+		return componentToAdd;
+	};
+
+	void AddComponent(Component* aComponent)
+	{
+		// Deduce type and add it to the correct component list
+		if (dynamic_cast<TransformComponent*>(aComponent) != nullptr)
+			myTransformComponents.push_back(static_cast<TransformComponent*>(aComponent));
+			
+		// if(dynamic_cast<\new type\>(aComponent)
+		//		\new type list\.push_back(static_cast<\new type\*>(aComponent));
+	}
```

Comment:
- Consider the use of the const modifier.
- You seem to be casting twice on the component in function AddComponent. Can't you same the return value of the dynamiccast and use it for the push_back?
- What happens when you try and add a class that isn't derived from Component or is the Component base class? You might want to consider adding some extra template magic such as std::enable_if & std::is_base_of<Type> to disable functions like these when the interface is improperly used by the coder/user. Only if you want to add the safety layer of course, might still be interesting.
https://www.cplusplus.com/reference/type_traits/enable_if/
https://en.cppreference.com/w/cpp/types/is_base_of

Reply:
- Considered to use const here, but for Entity I might still want to add it to a list in the entity as a member for AddComponent, although I think using it in CreateComponent is better.
- As static cast is compile time, it wouldn't make a difference in performance, but it might be more readable.
- Oh, I'll definitely still add safety checks to this!